### PR TITLE
Rerun blenderkit conversions

### PR DESCRIPTION
**Pull request by tomonorman**
Basically an "Optimize!" button which will go through every model originally from blendkit and rerun the conversion process _while keeping the original index.json_ (in case of any manual curation). This is so it can be run when the blendkit conversion logic has been updated / further optimized.

image

This should either be merged into the optimize-glb branch, or the dev branch after @azazdeaz optimize-glb PR has been merged.

### CODE REVIEW

**Review by azazdeaz — APPROVED** [review]
Thank you @tomonorman!!

File: roboprop_client/views.py [hunk 1]
```diff
@@ -544,3 +544,42 @@ def add_metadata(request, name):
...
         return redirect("mymodels")
+
+
+def update_models_from_blendkit(request):
+    if request.method == "POST":
+        # get index.json
+        index = _check_and_get_index(request)
+        for model in index:
+            # check if model has key assetBaseId, i.e is from blendkit
+            if "assetBaseId" in index[model]:
```
Comment (fabid) [inline roboprop_client/views.py:555]: Good as it is now, but maybe we will need a `source` field in the future?

**Review by fabid — APPROVED** [review]
Thank you. that will be useful

File: roboprop_client/views.py (hunk 1 above)
Comment (tomonorman) [reply roboprop_client/views.py:555]: `source` as in what library it originally came from? (own, fuel, blendkit etc etc.). Yes I would agree we most likely should have it in future. thank you

**Pull request merged**